### PR TITLE
Get rid of the last shim warning with IPython/Jupyter 4

### PR DESCRIPTION
**Pull request by stonebig**

### CODE REVIEW

File: spyderlib/widgets/sourcecode/codeeditor.py [hunk 1]
```diff
@@ -59,8 +59,12 @@
 from spyderlib.py3compat import to_text_string
 
 try:
-    import IPython.nbformat as nbformat
-    from IPython.nbconvert import PythonExporter as nbexporter
+    try:   # Ipython 4
+        import nbformat as nbformat
+        from nbconvert import PythonExporter as nbexporter  
+    except:  # Ipython 3
```
Comment (goanpeca) [inline spyderlib/widgets/sourcecode/codeeditor.py]: Thanks for this @stonebig 

But we should catch the actual exception, like `except ImportError?`

Also the comments

File: spyderlib/widgets/sourcecode/codeeditor.py [hunk 2]
```diff
@@ -59,8 +59,12 @@
...
 try:
-    import IPython.nbformat as nbformat
-    from IPython.nbconvert import PythonExporter as nbexporter
+    try:   # Ipython 4
+        import nbformat as nbformat
+        from nbconvert import PythonExporter as nbexporter  
+    except:  # Ipython 3
+        import IPython.nbformat as nbformat
+        from IPython.nbconvert import PythonExporter as nbexporter
 except:
```
Comment (Nodd) [inline spyderlib/widgets/sourcecode/codeeditor.py]: Same here...

**Comment by ccordoba12** [comment]
+1 to the changes suggested by @goanpeca and @Nodd. After that, this is ready to be merged.

**Comment by ccordoba12** [comment]
Thanks!

**Comment by stonebig** [comment]
You're welcomed.

**Pull request merged**